Automattic/simplenote-electron
# Fixing publish status not updating

Fixing #430 seemed trivial until I discovered that we weren't really setting the `note` state properly when the app would run for the first time. I fixed this by moving some logic that used to live in `render()` of `app.jsx` to the app state.

The main fix for the bug reported in #430 was to properly access the note object from the state in `share.jsx`.

**To Test**
* Launch the app, the first note should be selected.
* Click on the share button, the dialog should show properly.
* Select a note further down the list, and delete it. The previous note in the list should now be selected.
* Publish a note. When it is published, the UI should properly update with the short Url.
